packages/react/src/components/header/components/headerActionBar/HeaderActionBarNavigationMenu.tsx

```diff
@@ -513,12 +513,23 @@ export const HeaderActionBarNavigationMenu = ({
       id="hds-mobile-menu"
       onTransitionEnd={animationDone}
     >
-      {actionBarItems?.map?.((item: HeaderActionBarItemProps) => {
-        if (typeof item === 'object') {
-          return React.cloneElement(item as unknown as React.ReactElement, { fullWidth: true, key: getKey(item) });
-        }
-        return null;
-      })}
+      {actionBarItems && actionBarItems.length && (
```

Comment:
Hmm... would it work by just modifying the previous implementation with `{ mobileMenuOpen && actionBarItems?.map?.` and so on?

Reply:
actionBarItems is an empty array by default (created in ActionBar), so the wrapper would be rendered even if there are no items.

Reply:
So the wrapper is needed even if empty?

Reply:
I guess I didn't understand the question. 😄 

The new code makes sure the wrapper is **not** added, if there are zero items in the `actionBarItems`. The `actionBarItems` always exists, so length must be checked.

The `mobileMenuOpen` is not checked like it wasn't checked before. I think because menu needs to be rendered for open/close animations to work.

Reply:
Hmm... didn't see any difference after simpifying it to be like this:

```suggestion
    <div
      className={classNames(styles.headerNavigationMenu, mobileMenuOpen && styles.mobileMenuOpen)}
      id="hds-mobile-menu"
      onTransitionEnd={animationDone}
      aria-hidden={!mobileMenuOpen}
    >
      {actionBarItems?.map?.((item: HeaderActionBarItemProps) => {
        if (typeof item === 'object') {
          return React.cloneElement(item as unknown as React.ReactElement, {
            fullWidth: true,
            key: getKey(item),
          });
        }
        return null;
      })}
```

Reply:
⬆️  would also get rid of the need for the extra wrapper

Reply:
This got interesting...

Reverted back with simple "aria-hidden" in the parent. The hidden "Kirjaudu"-button got still focus with tab. There was a error in the console about "Chrome refusing to set aria-hidden to element with active items."

Reverted all changes and just set visibility hidden to the parent when menu is not open. It works now.

Reply:
Now it's working again and reviewable.